src/commands/malid.rs

```diff
@@ -1,28 +1,62 @@
+use chrono::NaiveDate;
 use serenity::builder::CreateApplicationCommand;
 use serenity::model::prelude::interaction::application_command::CommandDataOption;
 
 use serde::Deserialize;
 
 #[derive(Deserialize, Debug)]
 #[serde(rename_all = "PascalCase")]
-pub struct MenuItem {
-  title: String,
+pub struct DailyMenu {
   date: String,
-  vegan_menu: String,
+  #[serde(rename = "Title")]
+  main_course: String,
+  #[serde(rename = "VeganMenu")]
+  vegan_course: String,
   soup_of_the_day: String,
 }
 
-pub fn run(_options: &[CommandDataOption]) -> String {
-  "Hey, I'm alive!".to_string()
+fn format_week_menu(week_menu: Vec<DailyMenu>) -> String {
+  let mut output = String::new();
+  for menu in week_menu.iter() {
+    output += format_specific_day(menu).as_str();
+  }
+  return output;
+}
+
+fn format_specific_day(menu: &DailyMenu) -> String {
+  let date = NaiveDate::parse_from_str(&menu.date, "%Y-%m-%d").expect("Expected a date");
+
+  format!(
+    "## {}:\n\t🍴 {}\n\t🥬 {}\n\t🍲 {}\n",
+    date.format("%A, %-d %b"),
+    menu.main_course,
+    menu.vegan_course,
+    menu.soup_of_the_day
+  )
+}
+
+pub async fn run(_options: &[CommandDataOption]) -> Result<String, String> {
+  let week_menu = this_weeks_menu().await.expect("Some error happened");
```

Comment:
Same as above, expect panics and stops the program execution, let's try to keep it away from user-inputs :)